astropy/table/tests/test_table.py

```diff
@@ -1174,6 +1174,25 @@ def test_copy_protocol():
     _assert_copies(t, t2, deep=False)
     _assert_copies(t, t3)
 
+
+def test_max_rows_copy():
+    """
+    Test max_rows argument for copy method.
+    """
+    t = table.Table([[1, 2, 3], [2, 3, 4]], names=['x', 'y'])
+
+    tcopy_1 = t.copy()
+    assert len(tcopy_1) == len(t)
+
+    tcopy_2 = t.copy(max_rows=0)
+    assert len(tcopy_2) == 0
```

Comment:
Check here that the column names and types are the same.  We're pretty sure this works because slicing works, but since the intent of this PR is to allow for zero-length copying of a table then it should be explicitly checked here.